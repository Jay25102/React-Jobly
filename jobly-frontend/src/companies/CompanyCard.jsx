import React from "react";
import { Link } from "react-router-dom";
import "./CompanyCard.css";

/**
 * Displays info about companies
 * 
 * Rendered by CompanyList
 */
function CompanyCard({name, handle, description, logoUrl}) {
    console.debug("CompanyCard", logoUrl);

    return (
        <Link className="CompanyCard card" to={`/companies/${handle}`}>
            <div className="card-body">
                <h6 className="card-title">
                    {logoUrl && <img src={logoUrl} alt={name}/>}
                    {name}
                </h6>
                <p><small>{description}</small></p>
            </div>
        </Link>
    )
}

export default CompanyCard;