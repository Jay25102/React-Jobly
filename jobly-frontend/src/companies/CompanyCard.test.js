import React from "react";
import { render } from "@testing-library/react";
import CompanyCard from "./CompanyCard";
import { UserProvider } from "../testUtils";
import { MemoryRouter } from "react-router-dom";

describe("JobCard", function() {
    test("it renders without crashing", function() {
        render(
            <MemoryRouter>
                <CompanyCard
                    handle="handle"
                    name="name"
                    description="description here"
                    logoUrl="/logos/logo1.jpg"
                />                
            </MemoryRouter>
    );
    });
    test("it matches snapshot", function() {
        const { asFragment } = render(
            <MemoryRouter>
                <CompanyCard
                    handle="handle"
                    name="name"
                    description="description here"
                    logoUrl="/logos/logo1.jpg"
                />                
            </MemoryRouter>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
