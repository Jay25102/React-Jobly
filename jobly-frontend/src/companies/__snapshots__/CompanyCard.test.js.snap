// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobCard it matches snapshot 1`] = `
<DocumentFragment>
  <a
    href="/companies/handle"
  >
    <div>
      name
      <img
        alt="name"
        src="/logos/logo1.jpg"
      />
      description here
    </div>
  </a>
</DocumentFragment>
`;
